### PR TITLE
feat(product-review):A buyer can provide a review on a product

### PR DESCRIPTION
**Pull request by TuyishimireEric**
#### What does this PR do?

A buyer is able to provide feedback and rating on a product they

#### Description of Task to be completed?

> A buyer should be able to add a review on the product they have bought.
> 
> For now, the payment is not yet implemented but you need to find a way to leave that check for an other time.
> 
> For now, implement a logic for a rating system (0-5) together with a feedback that any buyer can put on any product.
> 
> then later once the payment if implemented you will add a check of whether the same buyer bought that product before.
> 

#### How should this be manually tested?

- run `npm install` 
- run `npm run migrate`
- run `npm run seed`
- run `npm run dev`
- check the [swagger documentation ](http://localhost:3000/swagger/)

#### What are the relevant pivotal tracker/Trello stories?

- id of the project #187300162

### CODE REVIEW

**Review by Mukunzijames — APPROVED** [review]

**Review by Ruthuwamahoro — APPROVED** [review]

File: src/controllers/categoryController.ts [hunk 1]
```diff
@@ -113,18 +113,6 @@ export const deleteCategory = async (req: Request, res: Response) => {
...
-    });
-
-    if (productsUsingCategory) {
-      return sendResponse<null>(
-        res,
-        400,
-        null,
-        'Cannot delete category as it is associated with one or more products.'
-      );
-    }
```
Comment (fab-ryan) [inline src/controllers/categoryController.ts:127]: @TuyishimireEric  why did you remove this lines of codes

**Review by fab-ryan — APPROVED** [review]

**Review by fab-ryan — APPROVED** [review]
